### PR TITLE
Update dependencies & Drop uuid-parse module

### PR DESCRIPTION
**Pull request by CodeMan99**
Also took the liberty of replace `new Buffer` with the suggested static methods.

### CODE REVIEW

File: package.json
Comment (jwhitlock) [inline package.json:15]: [The maintainers say](https://github.com/uuidjs/uuid#upgrading-from-uuid3):

> In order to avoid confusion with RFC [version 4](https://github.com/uuidjs/uuid#uuidv4options-buffer-offset) and [version 5](https://github.com/uuidjs/uuid#uuidv5name-namespace-buffer-offset) UUIDs, and a possible [version 6](http://gh.peabody.io/uuidv6/), releases 4 thru 6 of this module have been skipped.

So, this is not as extreme a version jump as it seems.

[Version 8.3.0](https://github.com/uuidjs/uuid/blob/master/CHANGELOG.md#830-2020-07-27) added ``uuid.parse`` and similar, partially in response to [uuid-parse](https://www.npmjs.com/package/uuid-parse) being marked as unmaintained.

File: slugid.js
Comment (jwhitlock) [inline slugid.js:31]: Future work: there is a [`base64url` encoding](https://nodejs.org/api/buffer.html#buffer_buffers_and_character_encodings) added in [node 15.7.0](https://nodejs.org/en/blog/release/v15.7.0/) that encodes directly into the RFC 4648 section 5 format.

File: slugid.js
Comment (jwhitlock) [inline slugid.js:23]: Nit: this could be:
```suggestion
const uuid = require('uuid');
```

It may be better to change all the imports in the future as part of converting to ``import`` syntax.

**Review by jwhitlock — APPROVED** [review]
Thanks @CodeMan99, this looks like a clean update.

The TravisCI integration is broken or too slow, so I checked it out and ran it locally. Tests pass under node v14.16.1.

File: slugid.js
Comment (CodeMan99) [reply slugid.js:23]: Seems a bit of a cherry-picked sore spot. There's plenty of minor improvements like this that could be made. Such as general `var` usage or the import/module pattern.

I'm happy to make the change if that helps get this PR merged. Personally, I wouldn't make the change to `const` without a proceeding `"use strict";` statement. The [travis-ci config](https://github.com/taskcluster/slugid/blob/main/.travis.yml#L7) claims to test against nodejs v4.

Comment (jwhitlock) [reply slugid.js:23]: For me, "nit" means it is a suggestion, often based on coding style, that the submitter can take or leave, and doesn't need to be changed for this PR to merge. I think it is OK to keep this change focused on the ``uuid`` upgrade, rather than chase down all the issues, and that it will be cleaner due to that focus.

The reason I didn't merge immediately is that I wanted to get some further feedback from Mozilla team members with more Taskcluster experience. I've gotten a little in other channels, and none that suggests changes or blocks merging. I'm happy to merge in a few hours.

> Personally, I wouldn't make the change to const without a proceeding "use strict"; statement. The travis-ci config claims to test against nodejs v4.

I think this module has gone a bit stale, since changes were unneeded. I think we should add current node versions to the support matrix, and consider dropping old node versions, as part of the next release. That's definitely outside the scope of this PR.

**Pull request merged**
